### PR TITLE
[flutter_tools] use correct sdk path for analysis

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

flutter analyze was unconditionally using the cached dart-sdk for analysis, and was not running with the built SDK during local engine. This broke when trying to update the analyzer for null safety, since it required us to wait for the dart change to roll into the framework first.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 1]
```diff
@@ -90,7 +92,8 @@ class AnalyzeOnce extends AnalyzeBase {
     final Completer<void> analysisCompleter = Completer<void>();
     final List<AnalysisError> errors = <AnalysisError>[];
 
-    final String sdkPath = argResults['dart-sdk'] as String ?? sdk.dartSdkPath;
+    final String sdkPath = argResults['dart-sdk'] as String ??
+      artifacts.getArtifactPath(Artifact.engineDartSdkPath);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart:96]: This was the actual bug fix, everything else is for testing

File: packages/flutter_tools/lib/src/commands/analyze_continuously.dart [hunk 2]
```diff
@@ -68,7 +70,8 @@ class AnalyzeContinuously extends AnalyzeBase {
       analysisTarget = fileSystem.currentDirectory.path;
     }
 
-    final String sdkPath = argResults['dart-sdk'] as String ?? sdk.dartSdkPath;
+    final String sdkPath = argResults['dart-sdk'] as String ??
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/analyze_continuously.dart:73]: And here too

**Review by jmagman — COMMENTED** [review]
[drive](https://github.com/flutter/flutter/blob/886313393fce737da812944aef4a6b94e703c6ee/packages/flutter_tools/lib/src/commands/drive.dart#L511) pulls the same thing.

**Comment by jonahwilliams** [comment]
I should probably move that API into the cache so it can't be misued

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**